kore/src/Kore/Rewrite/SMT/Translate.hs

```diff
@@ -533,6 +533,11 @@ backTranslateWith
         backTranslate (List xs)
             | null xs =
                 throwError "backtranslate: empty list"
+            -- special case for the unary minus, back-translating 'List ["-", "X"]' as '0 - X'
+            | [fct@(Atom "-"), arg] <- xs
+            , Just koreSym <- Map.lookup fct reverseMetadata = do
+                arg' <- backTranslate arg
+                pure $ TermLike.mkApplySymbol koreSym [mkInternalInt (InternalInt (simpleSort "SortInt") 0), arg']
```

Comment:
I think we should rather just negate the `InternalInt` that we expect inside `arg`, unless that does not work for some reason...
```suggestion
                arg' <- backTranslate arg
                case arg' of
                    InternalInt intSort n -> pure $ InternalInt intSort (negate n)
                    other -> throwError $ "unable to translate negation of " <> show other
```

Reply:
Thanks, that worked, modulo juggling Kore's data constructors.